CRM/Financial/BAO/FinancialType.php

```diff
@@ -181,11 +200,159 @@ public static function getIncomeFinancialType() {
+      ) {
         unset($financialType[$key]);
       }
     }
     return $financialType;
   }
 
+  /**
+   * adding permissions for financial types
+   *
+   *
+   * @param array $permissions
+   *   an array of permissions
+   */
+  public static function permissionedFinancialTypes(&$permissions) {
+    if (!self::isACLFinancialTypeStatus()) {
+      return FALSE;
+    }
+    $financialTypes = CRM_Contribute_PseudoConstant::financialType();
+    $prefix = ts('CiviCRM') . ': ';
+    foreach ($financialTypes as $id => $type) {
+      $permissions['add contributions of type ' . $type] = array(
+        $prefix . ts('add contributions of type ') . $type,
+        ts('Add contributions of type ') . $type,
+      );
+      $permissions['view contributions of type ' . $type] = array(
+        $prefix . ts('view contributions of type ') . $type,
+        ts('View contributions of type ') . $type,
+      );
+      $permissions['edit contributions of type ' . $type] = array(
+        $prefix . ts('edit contributions of type ') . $type,
+        ts('Edit contributions of type ') . $type,
+      );
+      $permissions['delete contributions of type ' . $type] = array(
+        $prefix . ts('delete contributions of type ') . $type,
+        ts('Delete contributions of type ') . $type,
+      );
+    }
+    $permissions['administer CiviCRM Financial Types'] = array(
+      $prefix . ts('administer CiviCRM Financial Types'),
+      ts('Administer access to Financial Types'),
+    );
+  }
+
+  public static function getAvailableFinancialTypes(&$financialTypes = NULL, $action = 'view', $resetCache = FALSE) {
+    if (empty($financialTypes)) {
+      $financialTypes = CRM_Contribute_PseudoConstant::financialType();
+    }
+    if (!self::isACLFinancialTypeStatus()) {
+      return $financialTypes;
+    } 
+    // check cached value
+    if (CRM_Utils_Array::value($action, self::$_availableFinancialTypes) && !$resetCache) {
+      $financialTypes = self::$_availableFinancialTypes[$action];
+      return self::$_availableFinancialTypes[$action];
+    }
+    foreach ($financialTypes as $finTypeId => $type) {
+      if (!CRM_Core_Permission::check($action . ' contributions of type ' . $type)) {
+        unset($financialTypes[$finTypeId]);
+      }
+    }
+    self::$_availableFinancialTypes[$action] = $financialTypes;
+    return $financialTypes;
+  }
+  
+  public static function getAvailableMembershipTypes(&$membershipTypes = NULL, $action = 'view') {
+    if (empty($membershipTypes)) {
+      $membershipTypes = CRM_Member_PseudoConstant::membershipType();
+    }
+    if (!self::isACLFinancialTypeStatus()) {
+      return $membershipTypes;
+    }
+    foreach ($membershipTypes as $memTypeId => $type) {
+      $finTypeId = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_MembershipType', $memTypeId, 'financial_type_id');
+      $finType = CRM_Contribute_PseudoConstant::financialType($finTypeId);
```

Comment:
Well, they still exist and they still work. The "new" way of doing this would be:

    $finType = CRM_Core_Pseudoconstant::getName('CRM_Member_DAO_MembershipType', 'financial_type_id', $finTypeId);